iterator.go

```diff
@@ -0,0 +1,23 @@
+package mmark
+
+import "bytes"
+
+// linespan implements a minimal line iterator over '\n' delimited content
+type linespan struct{ begin, end int }
+
+// next updates begin and end to point to the next line
+func (sc *linespan) next(content []byte) bool {

```

Comment:
Thanks for all changes. I however thing this add something new that should happen in another PR. This (`linespan`) is only used once (now), and there is is basically to prefix each line with the indent.